File: kakao2022_blind/hyunsoo/prime_num_in_k_number/PrimeNumInKNumber.java
package kakao2022_blind.prime_num_in_k_number;

class Solution {
    boolean check(long num) {
        if (num == 2)
            return true;
        if (num == 1 || num % 2 == 0)
            return false;

        for (long i = 3; i <= (long) Math.sqrt(num); i += 2) {
            if (num % i == 0)
                return false;
        }

        return true;
    }

    String convert(int num, int base) {
        int q = num / base, r = num % base;
        if (q != 0)
            return convert(q, base) + String.valueOf(r);
        else
            return String.valueOf(r);
    }

    public int solution(int n, int k) {
        String number = k == 10 ? String.valueOf(n) : convert(n, k);
        String[] nums = number.split("0+"); // 0 한 개 이상으로 구별, 빈 문자열 안 들어감

        int answer = 0;
        for (String value : nums) {
            if (check(Long.parseLong(value)))
                answer++;
        }
        return answer;
    }
}

public class PrimeNumInKNumber {
    public static void main(String[] args) {
        Solution so = new Solution();
        System.out.println(so.solution(437674, 3));
    }
}

File: kakao2022_blind/hyunsoo/prime_num_in_k_number/prime_num_in_k_number.py
# https://www.youtube.com/watch?v=qojmdTg0du4&t=7s
# k 진수에서 소수의
# 개수 구하기
import math

# 소수 인지 빠르게 판별


def check(num):
    if num == 2:
        return True
    if num == 1 or num % 2 == 0:
        return False

    for i in range(3, int(math.sqrt(num)) + 1, 2):
        if num % i == 0:
            return False

    return True

# base 진법으로 변환


def convert(num, base):
    q, r = divmod(num, base)
    if q:
        return convert(q, base) + str(r)
    else:
        return str(r)


def solution(n, k):
    numstr = str(n) if k == 10 else convert(n, k)
    nums = numstr.split('0')

    answer = 0
    for value in nums:
        # 빈 문자열("") 아니고 소수이면 카운팅
        if len(value) and check(int(value)):
            answer += 1

    return answer


print(solution(437674, 3))
# result 3

# print(solution(110011, 10))
# # result 2
